sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -251,6 +258,15 @@ size_t gnrc_lorawan_build_uplink(gnrc_lorawan_t *mac, iolist_t *payload,
     lw_hdr->addr = mac->dev_addr;
     lw_hdr->fctrl = 0;
 
+    lorawan_hdr_set_adr(lw_hdr,mac->mlme.adr);
+
+    /* Set `ADRACKReq` bit and ADR_ACK_CNT */
+    if (mac->mlme.adr) {
+        lorawan_hdr_set_adr_ack_req(lw_hdr,
```

Comment:
This snippet doesn't reflect what the standard specified.

From the standard:
```
In   uplink   transmissions   the ADRACKReqbit   is   set   if ADR_ACK_CNT  >=  ADR_ACK_LIMIT and  the  current  data-rate  is  greater than the  device  defined minimum data rate, it is cleared in other conditions
```

The bit shouldn't be set if the DR cannot be increased.

Reply:
Done.

Reply:
it still doesn't reflect the standard. The ADR ACK bit is not set if ADR_ACK_CNT < ADR_ACK_LIMIT.

Reply:
Okay, makes sense to move the condition one line up.